cpp/modmesh/buffer/SimpleArray.cpp

```diff
@@ -146,83 +146,76 @@ DataType get_data_type_from_type<double>()
     return DataType::Float64;
 }
 
-SimpleArrayPlex::SimpleArrayPlex(const shape_type & shape, DataType data_type)
+#define CREATE_SIMPLE_ARRAY(DataType, ArrayType, ...)                          \
```

Comment:
reduce a lot of code, and allow us to call different constructor of `SimpleArray`

Reply:
added comments and added `undef`